service/storage_service.cc

```diff
@@ -4400,8 +4400,8 @@ future<> storage_service::removenode(locator::host_id host_id, std::list<locator
 
                     // Step 7: Announce the node has left
                     slogger.info("removenode[{}]: Advertising that the node left the ring", uuid);
-                    // FIXME: lock_endpoint
-                    gms::permit_id pid;
+                    auto permit = ss._gossiper.lock_endpoint(endpoint, gms::null_permit_id).get();
+                    const auto& pid = permit.id();
```

Comment:
I can see that we're solving more than just reentrancy, we're making some previously separate operations appear as atomic.